modules/currency/currency.service.js

```diff
@@ -18,6 +18,17 @@ class CurrencyService {
   }
 
   async addCurrency(data) {
+    if (await this.checkCurrencyExist(data)) {
+      throw new Error(CURRENCY_ALREADY_EXIST);
+    }
+    return new Currency(data).save();
+  }
+
+  async deleteCurrency(id) {
+    return await Currency.findByIdAndDelete(id);
+  }
+
+  async checkCurrencyExist(data) {
     const currency = await Currency.find({
```

Comment:
Use the method `count` here to reduce the amount of data that is transferred between app and db